### PR TITLE
feat(events-targets): kinesis stream as event rule target

### LINKED ISSUE #2997: Support Kinesis Streams as an Event Rule target
* **I'm submitting a ...**
  - [ ] :beetle: bug report
  - [X] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/awslabs/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the expected behavior (or behavior of feature suggested)?**
CloudWatch Event Rules can send events to Kinesis Streams, as shown here https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-target.html#aws-properties-events-rule-target--examples--Target_with_KinesisParameters--yaml

The CDK does not yet support that, as shown here: https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-events.IRuleTarget.html

* **What is the motivation / use case for changing the behavior or adding this feature?**
I want to send CloudWatch Events to a Kinesis Stream.  This is needed for GuardDuty event collection as shown here: https://summitroute.com/blog/2019/03/06/guardduty_event_collection_via_cloudwatch_events/

* **Please tell us about your environment:**
  
  - CDK CLI Version: 0.35.0
  - Module Version: aws-events
  - OS: [all ]
  - Language: [ Javascript ]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

### PR DESCRIPTION
**Pull request by gcacace**
closes #2997

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   * result from a CloudWatch event.
+   */
+  public bind(_rule: events.IRule, _id?: string): events.RuleTargetConfig {
+    const policyStatements = [new iam.PolicyStatement({
+      actions: ['kinesis:PutRecord', 'kinesis:PutRecords'],
+      resources: [this.stream.streamArn],
+    })];
+
+    return {
+      id: '',
```
Comment (eladb) [inline packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts:54]: Does it make sense that the id is empty?

Comment (gcacace) [reply packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts:54]: I've mimicked what's already in other targets:
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/ecs-task.ts#L173
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/batch.ts#L68
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/codebuild.ts#L34
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/lambda.ts#L36
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/sns.ts#L42
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/sqs.ts#L65
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/state-machine.ts#L32

If it does not make sense, I would recommend to address this as separate pull request (and refactor all the existing targets accordingly).

Comment (eladb) [reply packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts:54]: Okay, thanks

File: packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { singletonEventRole } from './util';
+
+/**
+ * Customize the Kinesis Stream Event Target
+ */
+export interface KinesisStreamProps {
+  /**
+   * Partition Key Path for records sent to this stream
+   *
+   * @default eventId as the partition key
```
Comment (eladb) [inline packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts:13]:
```suggestion
   * @default - eventId as the partition key
```

File: packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   * @default - eventId as the partition key
+   */
+  readonly partitionKeyPath?: string;
+
+  /**
+   * The message to send to the stream.
+   *
+   * Must be a valid JSON text passed to the target stream.
+   *
+   * @default the entire CloudWatch event
```
Comment (eladb) [inline packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts:22]:
```suggestion
   * @default - the entire CloudWatch event
```

**Pull request merged**